assembly/src/assembler/instruction/adv_ops.rs

```diff
@@ -14,7 +15,16 @@ use crate::{ADVICE_READ_LIMIT, AssemblyError};
 /// Returns an error if the specified number of values to pushed is smaller than 1 or greater
 /// than 16.
 pub fn adv_push(block_builder: &mut BasicBlockBuilder, n: u8) -> Result<(), AssemblyError> {
-    validate_param(n, 1..=ADVICE_READ_LIMIT)?;
+    let min = 1;
+    let max = ADVICE_READ_LIMIT;
+
+    if n < min || n > max {
+        return Err(AssemblyError::Other(
+            miette!("Invalid parameter: n = {} is out of range (expected {}..={})", n, min, max)
+                .into(),
+        ));
```

Comment:
Let's use `AssemblyError::InvalidU8Param` similar to the others (with the `span` and `source_file` properly populated from the `ProcedureContext`)